prov/efa/src/efa_mr.c

```diff
@@ -511,76 +521,67 @@ struct ibv_mr *efa_mr_reg_ibv_dmabuf_mr(struct ibv_pd *pd, uint64_t offset,
 static struct ibv_mr *efa_mr_reg_ibv_mr(struct efa_mr *efa_mr, struct fi_mr_attr *mr_attr,
 					int access, const uint64_t flags)
 {
-	if (flags & FI_MR_DMABUF)
-		return efa_mr_reg_ibv_dmabuf_mr(
-			efa_mr->domain->ibv_pd,
-			mr_attr->dmabuf->offset,
-			mr_attr->dmabuf->len,
-			(uintptr_t) mr_attr->dmabuf->base_addr + mr_attr->dmabuf->offset,
-			mr_attr->dmabuf->fd,
-			access
-		);
+	int ret, dmabuf_fd;
+	struct ibv_mr *mr = NULL;
+	uint64_t iova, offset;
+	size_t len;
+	enum efa_dmabuf_support_status *dmabuf_support;
+
+	dmabuf_support = &(efa_mr->domain->hmem_info[mr_attr->iface].dmabuf_support_status);
+
+	if (flags & FI_MR_DMABUF) {
+		goto dmabuf;
```

Comment:
This should just return the ibv_reg_dmabuf_mr call, we don't need any fallback here.